cmd/osbuild-service-maintenance/aws.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/semaphore"
+
+	"github.com/osbuild/osbuild-composer/internal/cloud/awscloud"
+)
+
+func AWSCleanup(maxConcurrentRequests int, dryRun bool, accessKeyID, accessKey, region string, cutoff time.Time) error {
+	a, err := awscloud.New(region, accessKeyID, accessKey, "")
+	if err != nil {
+		return err
+	}
+
+	var wg sync.WaitGroup
+	sem := semaphore.NewWeighted(int64(maxConcurrentRequests))
+	images, err := a.DescribeImagesByTag("Name", "composer-api-*")
```

Comment:
You're right, currently we don't actually expose that via `image-builder-crc` (the names are always generated).

We should tag them with something though, so we know these are images created by the service. I'll add that.

Reply:
You know i'm not sure if I'm actually a fan anymore, at best I can think of tagging it with the originating bucket, but that's about it, any other tag would be just as arbitrary but just slightly less likely to break if we do ever rename images.

Reply:
This is where I'm not that familiar with the AWS API. I guess there is not an option to list all images we built as there is in GCP, right? So we need to filter all images available to our account for those created by us (determined by a tag)? If there is no other easy way to get the appropriate list of images, other than filtering by tags, we could just tag every image with an **additional** tag which would be the same for all images.

If we don't expose the ability to specify the image name to the API users, the I think we can keep the code as is. However it would be IMO good to at least put a comment into the code describing the potential issue, so that we don't need to figure it out once again in the future.

Reply:
I'll document it then. We should consider adding an additional tag but I don't wanna just pick something random.

Reply:
Sounds good 🙂   Also I can pick something random for you 😄   as I did in https://github.com/osbuild/osbuild-composer/blob/24b83738f469e56b1c1bd192764f763e1c94c34d/internal/cloud/gcp/storage.go#L18-L24